core/src/main/java/org/elasticsearch/action/percolate/PercolateShardResponse.java

```diff
@@ -78,39 +72,27 @@ public PercolateShardResponse(BytesRef[] matches, List<Map<String, HighlightFiel
         }
     }
 
-    public PercolateShardResponse(BytesRef[] matches, long count, float[] scores, PercolateContext context, ShardId shardId) {
-        this(matches, EMPTY_HL, count, scores, context, shardId);
-    }
-
-    public PercolateShardResponse(BytesRef[] matches, List<Map<String, HighlightField>> hls, long count, PercolateContext context, ShardId shardId) {
-        this(matches, hls, count, EMPTY_SCORES, context, shardId);
-    }
-
-    public PercolateShardResponse(long count, PercolateContext context, ShardId shardId) {
-        this(EMPTY_MATCHES, EMPTY_HL, count, EMPTY_SCORES, context, shardId);
-    }
-
-    public PercolateShardResponse(PercolateContext context, ShardId shardId) {
-        this(EMPTY_MATCHES, EMPTY_HL, 0, EMPTY_SCORES, context, shardId);
+    public PercolateShardResponse(int count, PercolateContext context) {
+        this(new TopDocs(count, Lucene.EMPTY_SCORE_DOCS, 0f), Collections.emptyMap(), Collections.emptyMap(), context);
     }
 
-    public BytesRef[] matches() {
-        return matches;
+    public PercolateShardResponse(PercolateContext context) {
+        this(Lucene.EMPTY_TOP_DOCS, Collections.emptyMap(), Collections.emptyMap(), context);
     }
 
-    public float[] scores() {
-        return scores;
+    public TopDocs topDocs() {
+        return topDocs;
     }
 
-    public long count() {
-        return count;
+    public Map<Integer, String> ids() {
+        return ids;
     }
 
     public int requestedSize() {
         return requestedSize;
     }
 
-    public List<Map<String, HighlightField>> hls() {
+    public Map<Integer, Map<String, HighlightField>> hls() {
```

Comment:
can you add javadocs to say that eg. the integer key of the map represents a doc ID?